### PR TITLE
Increase step function test coverage - Part 2

### PR DESCRIPTION
**Pull request by JonTheBurley**
This PR adds tests and bug fixes for the graphRenderer, mainly focusing on getX and renderVertex. There is still much room for more coverage, but this PR was starting to get long so I stopped here for now.

Some bugs that were fixed:

- Removed excessive space between nodes within a Parallel box
- Nodes that can overlap (specifically a node after a Map since they are wider) now more intelligently position further away
- Maps drawn in the top-left of a Parallel won't default to the center of the screen now
- Parallel nodes that aren't part of a choice now render by parent correctly
- Nested Map and Parallel nodes now have correct Y depth values (the adjusting function was not recursive so only the top level of anything was moved up)

### CODE REVIEW

**Review by adorack-tri — APPROVED** [review]

File: src/StepFunctionGraph/GraphUtil.ts [hunk 1]
```diff
@@ -182,7 +182,10 @@ export const adjustDepthMatrix = (matrix: number[][], graph: Digraph): number[][
                 );
 
                 accum = accum.concat(outDegree).filter(v => v !== nextVertex);
-                removalArray = removalArray.concat(outDegree).filter(v => v !== nextVertex);
+
```
Comment (amirkibbar) [inline src/StepFunctionGraph/GraphUtil.ts:185]: Nit: this function has grown a lot, and it's a little difficult to follow the logic. Could you break it down to a few functions and call the "high level algorithm" from this function to make it more readable?

**Review by amirkibbar — APPROVED** [review]

File: src/StepFunctionGraph/GraphUtil.ts (hunk 1 above)
Comment (JonTheBurley) [reply src/StepFunctionGraph/GraphUtil.ts:185]: I agree it's long and hard to follow. It's easily the most complicated function in the renderer because it just requires that much data to get an x value. I'll keep the request in mind, but I'm not sure I see anywhere to break it up at a glance without passing in a lot of params. I've been considering breaking the way to get data from the drawn map and Digraph into a simpler function, which would free up a lot of the line bloat. I'll try to tackle this if I have time for it.

Comment (JonTheBurley) [reply src/StepFunctionGraph/GraphUtil.ts:185]: There is also the getX function that could use some breaking out of the drawn and Digraph data retrieval which other functions can benefit from and reduce line bloat.
